content/doc/book/using/best-practices.adoc

```diff
@@ -0,0 +1,177 @@
+
+=== Report build results
+
+Charts and graphs show trends and patterns that can highlight project status and progress.
+Automated test results for unit tests, integration tests, and end-to-end tests can highlight brittleness or instability.
+Coverage reports can illustrate areas where automated tests are not executing.
+Compiler warning messages can be the first indication of a problem.
+Static analysis tools can report risky code or code that may have security risks.
+Performance test results can identify delays or problem areas.
+
+The plugin:warnings-ng[Warnings Next Generation] plugin provides convenient access to many reports, including:
+
+* Compiler warnnings and errors (like gcc, clang, javac, or  golang)
+* Static analysis warnings and errors (spotbugs, checkstyle, pmd, lint, cpd, or Simian)
+* Code coverage reports
+
+.How to use the Warnings Next Generation plugin
+video::tj3xYFA6Q2o[youtube,width=800,height=420]
+
+=== Build on agents
+
+Use agents to perform builds instead of running builds on the controller.
+It is much safer and more scalable to use agents.
+
+More details are available in the link:/doc/book/security/controller-isolation/[controller isolation] documentation.
+
+=== Show failures to the right people
+
+Configure the notifications from failing and unstable jobs so the notifications reach the right people without distracting others.
+Many Jenkins users prefer to only be notified when a failure is likely their fault.
+They correctly assume that if they did not cause the failure, they are not the best person to investigate the failure.
+
+Refine your notification system so that new test failures first notify the most recent committers, since they are the most likely cause of the issue.
+
+.Sending Slack notifications
+video::EDVZli8GdUM[youtube,width=800,height=420]
+
+=== Use simple project names
+
+Jenkins uses project names for folders related to the project.
+Many tools cannot handle spaces, dollar signs, or similar characters in file paths.
+Limit yourself to alphanumeric characters [a-zA-Z0-9_-]+ in project names.
+Use the *Display Name* feature to make them look nice.
+The "Restrict project naming" setting in the system configuration can enforce naming restrictions for all projects.
+
+=== Fingerprint your dependencies
+
+When you have interdependent projects, it often becomes hard to keep track of which version of this is used by which version of that.
+Jenkins supports "file fingerprinting" to simplify this.
+
+More information is available on the link:/doc/book/using/fingerprints/[fingerprinting page].
+
+=== Don't use the Maven job type
+
+Jenkins has provided the plugin:maven-plugin[Maven integration plugin] for many years.
+Jobs created with the Maven integration plugin are created with the "Maven project" selection from the Jenkins "New item" menu.
+The Maven job type provides a higher level of integration with maven builds, but creates unnecessary complications due to that higher level of integration.
```

Comment:
```suggestion
While the Maven job type offers a higher level of integration with Maven builds, it can sometimes introduce unnecessary complexities due to this deep integration.
```